internal/internal_versioning_client.go

```diff
@@ -0,0 +1,319 @@
+// accounted for reachability as server cannot know if they'll happen as they do not use
+// assignment rules of their task queue. Same goes for Child Workflows or Continue-As-New Workflows
+// who inherit the parent/previous workflow's Build ID but not its task queue. In those cases, make
+// sure to query reachability for the parent/previous workflow's task queue as well.
+type BuildIDTaskReachability int
+
+const (
+	// BuildIDTaskReachabilityUnspecified indicates that task reachability was not reported.
+	BuildIDTaskReachabilityUnspecified = iota
+	// BuildIDTaskReachabilityReachable indicates that this Build ID may be used by new workflows or activities
+	// (based on versioning rules), or there are open workflows or backlogged activities assigned to it.
+	BuildIDTaskReachabilityReachable
+	// BuildIDTaskReachabilityClosedWorkflowsOnly specifies that this Build ID does not have open workflows
+	// and is not reachable by new workflows, but MAY have closed workflows within the namespace retention period.
+	// Not applicable to activity-only task queues.
+	BuildIDTaskReachabilityClosedWorkflowsOnly
+	// BuildIDTaskReachabilityUnreachable indicates that this Build ID is not used for new executions, nor
+	// it has been used by any existing execution within the retention period.
+	BuildIDTaskReachabilityUnreachable
+)
+
+type (
+	// TaskQueueVersionSelection is a task queue filter based on versioning.
+	// It is an optional component of [DescribeTaskQueueEnhancedOptions].
+	TaskQueueVersionSelection struct {
+		// Include specific Build IDs.
+		BuildIDs []string
+		// Include the unversioned queue.
+		Unversioned bool
+		// Include all active versions. A version is active if it has had new
+		// tasks or polls recently.
+		AllActive bool
+	}
+
+	// DescribeTaskQueueEnhancedOptions is the input to [Client.DescribeTaskQueueEnhanced].
+	DescribeTaskQueueEnhancedOptions struct {
+		// Name of the task queue. Sticky queues are not supported.
+		TaskQueue string
+		// An optional queue selector based on versioning. If not provided,
+		// the result for the default Build ID will be returned. The default
+		// Build ID is the one mentioned in the first unconditional Assignment Rule.
+		// If there is no default Build ID, the result for the
+		// unversioned queue will be returned.
+		Versions *TaskQueueVersionSelection
+		// Task queue types to report info about. If not specified, all types are considered.
+		TaskQueueTypes []TaskQueueType
+		// Include list of pollers for requested task queue types and versions.
+		ReportPollers bool
+		// Include task reachability for the requested versions and all task types
+		// (task reachability is not reported per task type).
+		ReportTaskReachability bool
+	}
+
+	// WorkerVersionCapabilities includes a worker's build identifier
+	// and whether it is choosing to use the versioning feature.
+	// It is an optional component of [PollerInfo].
+	WorkerVersionCapabilities struct {
+		BuildID       string
```

Comment:
nit: I know it is annoying but we should add GoDocs for this

Reply:
Thanks, I missed that one.